lib/taurus/qt/qtgui/panel/taurusmodelchooser.py

```diff
@@ -29,24 +29,120 @@
 from __future__ import print_function
 from __future__ import absolute_import
 
+from builtins import bytes, str
+
 import sys
-from taurus.external.qt import Qt
+import pkg_resources
+
+import taurus
+from taurus.external.qt import Qt, QtCore
 import taurus.core
 from taurus.qt.qtgui.container import TaurusWidget
 from taurus.qt.qtgui.tree import TaurusDbTreeWidget
 from taurus.core.util.containers import CaselessList
 from .taurusmodellist import TaurusModelList
 
+__all__ = ["TaurusModelSelectorTree", "TaurusModelChooser",
+           "TaurusModelSelector", "TaurusModelSelectorItem",
+           "TangoModelSelectorItem"]
+
+
+class TaurusModelSelector(Qt.QTabWidget):
+    """TaurusModelSelector is a QTabWidget container for
+    TaurusModelSelectorItem.
+    """
+    # TODO add action to add new TaurusModelSelectorItem
+    # TODO add mechanism to allow manual plugin activation
+    #      (instead of relying on installation)
+    modelsAdded = Qt.pyqtSignal('QStringList')
+
+    def __init__(self, parent=None):
+        Qt.QTabWidget.__init__(self, parent=parent)
+
+        self.currentChanged.connect(self.__setTabItemModel)
+        # ---------------------------------------------------------------------
+        # Note: this is an experimental feature
+        # It may be removed or changed in future releases
+        # Discover the taurus.modelselector plugins
+        ep_name = 'taurus.qt.qtgui.panel.TaurusModelSelector.items'
+        for ep in pkg_resources.iter_entry_points(ep_name):
+            try:
+                ms_class = ep.load()
+                ms_item = ms_class(parent=self)
+                self.__addItem(ms_item, ep.name)
 
-__all__ = ["TaurusModelSelectorTree", "TaurusModelChooser"]
-
+            except Exception as e:
+                err = 'Invalid TaurusModelSelectorItem plugin: {}\n{}'.format(
+                    ep.module_name, e)
+                taurus.warning(err)
+        # ---------------------------------------------------------------------
+
+    def __setTabItemModel(self):
+        w = self.currentWidget()
+        if w.model == "":
+            self.setCursor(QtCore.Qt.WaitCursor)
+            print(w)
+            w.setModel(w.default_model)
+            self.setCursor(QtCore.Qt.ArrowCursor)
```

Comment:
There are several things I would improve in `__setTabItemModel`:
- Do no not assume it  has a model (I know that right now you assume that it is a TaurusBaseComponent, but this limitation is probably not really needed)
- Consider the possibility of model being `None`
- Do not  assume that the cursor was an arrow before.  Instead, store the actual cursor before changing it to WaitCursor and restore it. 
- Do not do a print.
- It should be protected agains exceptions when setting the model 
Here is my proposed alternative:

```python
def __setTabItemModel(self):
    w = self.currentWidget()
    c = self.cursor()
    try:
        if not w.model:
            self.setCursor(QtCore.Qt.WaitCursor)
            w.setModel(w.default_model)
    except Exception as e:
        taurus.warning('Problem setting up selector: %r', e)
    finally:
        w.setCursor(c)
```

Reply:
Done